www/md_base/guanlecoja/config.coffee

```diff
@@ -88,4 +88,33 @@ gulp.task 'icons', ->
         ))
         .pipe(gulp.dest(path.join(config.dir.build, 'icons')))
 
+gulp.task 'apiproxy', ->
+    # this is a task for developing, it proxy api request to http://nine.buildbot.net
+    argv = require('minimist')(process.argv)
+    argv.apiserver ?= 'nine.buildbot.net'
+    argv.apiport ?= 8021
+    argv.server ?= 'http://localhost:8020'
+
+    fs = require 'fs'
+    path = require 'path'
+    http = require 'http'
+    httpProxy = require 'http-proxy'
+    proxy = httpProxy.createProxyServer({})
+    proxy.on 'proxyReq', (proxyReq, req, res, options) ->
+        delete proxyReq.removeHeader('Origin')
+        delete proxyReq.removeHeader('Referer')
+    proxy.on 'proxyRes', (proxyRes, req, res) ->
+        proxyRes.headers['Access-Control-Allow-Origin'] = '*'
+        console.log "[Proxy] #{req.method} #{req.url}"
+
+    server = http.createServer (req, res) ->
+        if req.url.match /^\/api/
+            proxy.web req, res, {target: 'http://' + argv.apiserver}
+        else if req.url.match /^\/ws/
+            proxy.ws req, res, {target: 'ws://' + argv.apiserver}
+        else
+            proxy.web req, res, {target: argv.server}
```

Comment:
I think the serve could serve directly the static files here. This will simplify everything.

You can add a script called

buildbot getwwwplugins
that would output such json as:
{
   "base": "/path/to/buildbot/www/md-base/static"
   "waterfall": "/path/to/buildbot/www/waterfall/static"
}

You can look at : 
https://github.com/buildbot/buildbot/blob/master/master/buildbot/scripts/runner.py#L634
https://github.com/buildbot/buildbot/blob/master/master/buildbot/scripts/dataspec.py
on how to create a new buildbot script entrypoint.

You can look at:
https://github.com/buildbot/buildbot/blob/master/master/contrib/api_proxy.py#L22
to see how to get the list of plugins.